xdsl/transforms/arith_add_fastmath.py

```diff
@@ -0,0 +1,90 @@
+"""Passes to manipulate fastmath flags in FP arith operations."""
+
+from dataclasses import dataclass
+from typing import Literal
+
+from xdsl.dialects import arith, builtin, llvm
+from xdsl.ir import Operation
+from xdsl.passes import MLContext, ModulePass
+from xdsl.pattern_rewriter import (
+    GreedyRewritePatternApplier,
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+)
+
+
+def _match_string_to_enum(flag: str, enum_type: type[llvm.FastMathFlag]):
+    for member in enum_type:
+        if flag == str(member.value):
+            return member
+
+    raise ValueError(f"{flag} is not a valid fastmath flag.")
+
+
+def _get_flag_list(flags: list[str], enum_type: type[llvm.FastMathFlag]):
+    list_flags: list[enum_type] = []
+    for flag in flags:
+        list_flags.append(_match_string_to_enum(flag, llvm.FastMathFlag))
+    return list_flags
+
+
+@dataclass
+class AddArithFastMathFlags(RewritePattern):
+    """Adds fastmath flags to FP binary operations from arith dialect."""
+
+    arith_op_cls: type[arith.FloatingPointLikeBinaryOp]
+    fastmath_op_attr: arith.FastMathFlagsAttr
+
+    def match_and_rewrite(self, op: Operation, rewriter: PatternRewriter) -> None:
+        if not isinstance(op, self.arith_op_cls):
+            return
+
+        op.fastmath = self.fastmath_op_attr
+
+
+@dataclass
+class AddArithFastMathFlagsPass(ModulePass):
+    """Module pass that adds fastmath flags to FP binary operations from arith dialect.
+    It currently does not preserve any existing fastmath flags that may already be part
+    of the operation.
+    By default (no arguments) it adds the "fast" flag.
+
+    Arguments (all optional):
+
+    - flags: {"fast", "none"} | list[str]: Set specific fastmath flags. Using "fast" or
+      "none" enables or disables all flags, respectively.
+    """
+
+    name = "arith-add-fastmath"
+
+    flags: Literal["fast", "none"] | list[str] = "fast"
+
+    def apply(self, ctx: MLContext, op: builtin.ModuleOp) -> None:
+        fm_flags = arith.FastMathFlagsAttr("fast")
+
+        if isinstance(self.flags, str):
+            fm_flags = arith.FastMathFlagsAttr(self.flags)
+        else:
+            if "none" in self.flags or "fast" in self.flags:
+                raise ValueError(
```

Comment:
I'm not sure if this needs to be tested; I looked at other passes which raise exceptions and we don't do it.
Not sure why.

Reply:
I guess because most passes we wrote were written in a paper rush :thinking: 
I feel that this logic should rather live in `FastMathFlagsAttr` either way; and would be more sensible to test there.

Reply:
I'll see if this warrants a separate PR, thanks!

Reply:
I had a look and the definition of `__init__` restricts the choices well. Moreover, any pass exceptions are not tested anywhere, so I'll just leave it as is for now.

This is the only point where string values coming from a user need to match to this enum.